easybuild/tools/filetools.py

```diff
@@ -259,75 +259,33 @@ def download_file(filename, url, path):
-        if download_file.last_time + 10 < time.time():
-            newblocks = blocks_read - download_file.last_block
-            download_file.last_block = blocks_read
-            tot_time = time.time() - download_file.last_time
-
-            if filesize <= 0:
-                # content length isn't always set
-                report_msg = "downloaded in %ss" % tot_time
-            else:
-                percent = blocks_read * blocksize * 100 // filesize
-                report_msg = "of %d kb downloaded in %ss [%d %%]" % (filesize / 1024.0, tot_time, percent)
-
-            downloaded_kbs = (blocks_read * blocksize) / 1024.0
-            kbps = (blocksize * newblocks) / 1024  // tot_time
-            _log.info("Download report: %d kb %s (%d kbps)", downloaded_kbs, report_msg, kbps)
-
-            download_file.last_time = time.time()
-
     # try downloading, three times max.
     downloaded = False
     attempt_cnt = 0
     while not downloaded and attempt_cnt < 3:
-        # get HTTP response code first before downloading file
-        response_code = None
         try:
-            urlfile = urllib.urlopen(url)
-            if hasattr(urlfile, 'getcode'):  # no getcode() in Py2.4 yet
-                response_code = urlfile.getcode()
-            urlfile.close()
-        except IOError, err:
-            _log.warning("Failed to get HTTP response code for %s, retrying: %s", url, err)
-
-        if response_code is not None:
-            _log.debug('HTTP response code for given url: %d', response_code)
-            # check for a 4xx response code which indicates a non-existing URL
-            if response_code // 100 == 4:
-                _log.warning('url %s was not found (HTTP response %d), not trying again', url, response_code)
-                return None
-
-        # use this functions's scope for variables we share with inner function used as report hook for urlretrieve
-        download_file.last_time = time.time()
-        download_file.last_block = 0
-
-        httpmsg = None
-        try:
-            (_, httpmsg) = urllib.urlretrieve(url, path, reporthook=report)
+            src_fd = urllib2.urlopen(url)
+            _log.debug('HTTP response code for given url: %d', src_fd.getcode())
+            write_file(path, src_fd.read())
             _log.info("Downloaded file %s from url %s to %s", filename, url, path)
-
-            if httpmsg.type == "text/html" and not filename.endswith('.html'):
-                _log.warning("HTML file downloaded to %s, so assuming invalid download, retrying.", path)
-                remove_file(path)
-            else:
-                # successful download
-                downloaded = True
-        except IOError, err:
-            _log.warning("Error when downloading from %s to %s (%s), removing it and retrying", url, path, err)
-            remove_file(path)
-
-        if not downloaded:
+            downloaded = True
+            src_fd.close()
+        except (ValueError, ) as err:
```

Comment:
@serverhorror: I threw out this block in the fork of your PR I'm working on (my PR is coming up soon)

if I get it right, this will only occur when a faulty download URL was passed (see example below)

also, the `shutil.copy` used in this block is unguarded (not in a try-except), so we'd need another level of try-except to handle any exceptions that may occur there...

why did you add this, did you run into a particular case where this was occurring, maybe in combination with a proxy setup?

```
>>> urllib2.urlopen('foo')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 127, in urlopen
    return _opener.open(url, data, timeout)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 396, in open
    protocol = req.get_type()
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib2.py", line 258, in get_type
    raise ValueError, "unknown url type: %s" % self.__original
ValueError: unknown url type: foo
```